### PR TITLE
Use `es5` as the TypeScript target

### PR DESCRIPTION
**Pull request by blakeembrey**
Closes https://github.com/buzinas/tslint-eslint-rules/issues/124.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,3 +2,5 @@ language: node_js
 
 node_js:
   - '6'
+  - '4'
+  - '0.10'
```
Comment (buzinas) [inline .travis.yml]: I'll never support Node v0.10 in this project (never did, btw).

File: package.json [hunk 2]
```diff
@@ -24,6 +24,7 @@
   "author": "Vitor Buzinaro <buzinas@buzinas.com>",
   "license": "MIT",
   "devDependencies": {
+    "any-promise": "^1.3.0",
```
Comment (buzinas) [inline package.json]: Please, change `any-promise` to `es6-promise` (it already comes with `.d.ts` and you can use ES6 imports.

File: src/readme/fetch.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import Promise = require('any-promise');
```
Comment (buzinas) [inline src/readme/fetch.ts]: Please, change to ES6 modules and `es6-promise`.

File: src/rules/validJsdocRule.ts [hunk 4]
```diff
@@ -198,12 +198,18 @@ class ValidJsdocWalker extends Lint.SkippableTokenAwareRuleWalker {
...
     }
 
-    const fn = this.fns.find(f => node === f.node);
+    let fn: IReturnPresent;
     let params = {};
     let hasReturns = false;
     let hasConstructor = false;
     let isOverride = false;
 
+    for (const f of this.fns) {
```
Comment (buzinas) [inline src/rules/validJsdocRule.ts]: There is no need for this `for...of`, you can do:

```javascript
const fn = this.fns.filter(f => node === f.node)[0];
```

File: package.json (hunk 2 above)
Comment (blakeembrey) [inline package.json]: So does `any-promise`, I contributed to both of these projects?

File: .travis.yml (hunk 1 above)
Comment (blakeembrey) [inline .travis.yml]: Oh, that's a bummer. I might just fork then. Thanks 😄

**Comment by blakeembrey** [comment]
Fixed the mentioned issues.

File: .travis.yml (hunk 1 above)
Comment (buzinas) [inline .travis.yml]: Sorry for that. Node 0.10 [will not be LTS by the end of this month](https://github.com/nodejs/LTS#lts-schedule), so there is no need to complicate even further this project. If you prefer to maintain a fork, I can understand that.

Comment (blakeembrey) [inline .travis.yml]: Yeah, I just support it in too many places and removing that support is a major breaking change for all my projects - projects like Typings.

Comment (buzinas) [inline .travis.yml]: It's weird that it didn't break anything before, because since the first commit Node v0.10 wasn't supported

Comment (blakeembrey) [inline .travis.yml]: In terms of features, 0.10 and 4.0 are pretty similar. It's only syntax really, which was handled by transpiling with TypeScript anyway.

Comment (buzinas) [inline .travis.yml]: I see... Well, if you think you can make the CI pass through v0.10, feel free to open another PR :smiley:

Comment (blakeembrey) [inline .travis.yml]: It honestly would be super helpful for me. I would love to not do a hostile fork and continue contributing to the same source instead.

Comment (buzinas) [inline .travis.yml]: I wouldn't call it hostile, but if it's helpful for you, and there are no problems for me - why not?

Comment (blakeembrey) [inline .travis.yml]: Maintenance becomes a pain. If you aren't supporting it, I would have no reason to contribute back here, which would result in me working on a mostly separate project at that point (and re-publishing to NPM under a different name). If I'm working on my own fork, I'd probably just start changing the style/development practices to suit me also. The project would look a lot different after I played with it on my own for a week...

**Pull request merged**